asmcomp/cmm.ml

```diff
@@ -261,3 +261,41 @@ let rec map_tail f = function
       cmm
   | c ->
       f c
+
+
+let map_shallow f = function
+  | Cconst_int _
+  | Cconst_natint _
+  | Cconst_float _
+  | Cconst_symbol _
+  | Cconst_pointer _
+  | Cconst_natpointer _
+  | Cblockheader _
+  | Cvar _
+    as c
+    ->
+      c
+
+  | Clet (id, e1, e2) ->
+      Clet (id, f e1, f e2)
+  | Cphantom_let (id, de, e) ->
+      Cphantom_let (id, de, f e)
+  | Cassign (id, e) ->
+      Cassign (id, f e)
+  | Ctuple el ->
+      Ctuple (List.map f el)
+  | Cop (op, el, dbg) ->
+      Cop (op, List.map f el, dbg)
+  | Csequence (e1, e2) ->
+      Csequence (f e1, f e2)
+  | Cifthenelse(cond, dbg_cond, e1, dbg_e1, e2, dbg_e2) ->
+      Cifthenelse (f cond, dbg_cond, f e1, dbg_e1, f e2, dbg_e2)
```

Comment:
If I remember correctly, the fields of `Cifthenelse` are usually labelled `(cond, ifso_dbg, ifso, ifnot_dbg, ifnot, dbg)`. Ignoring the names themselves, this suggests that it's not clear which debug info correspond to which expression, and it should be made explicit in the type definition (either by grouping expressions with their associated debug info as a tuple, or using an inline record).
This is orthogonal to this pull request, of course.

Reply:
(Tuples have my preference, or any other mean of grouping related data together.)

Reply:
I have a patch to tidy up this type along the same lines as the `Lambda.block` construct under discussion elsewhere.

Reply:
> This is orthogonal to this pull request, of course.

The choice of variable names was harmless but confusing.  Fixed!